dev/src/Split/Command.php

```diff
@@ -0,0 +1,253 @@
+use GuzzleHttp\Client;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * A Symfony command for executing subtree splits.
+ *
+ * @internal
+ */
+class Command extends GoogleCloudCommand
+{
+    const PARENT_TAG_NAME = 'https://github.com/googleapis/google-cloud-php/releases/tag/%s';
+    const EXEC_DIR = '.split';
+
+    private $components;
+    private $github;
+    private $split;
+
+    /**
+     * @param string $rootPath The path to the repository root directory.
+     * @param ComponentManager $components An instance of the Component Manager.
+     * @param GitHub|null $github The Github API wrapper. If not provided, it
+     *        will be instantiated. Available for testing purposes.
+     * @param Split|null $split The Splitsh wrapper. If not provided, it will be
+     *        instantiated. Available for testing purposes.
+     */
+    public function __construct(
+        $rootPath,
+        ComponentManager $components,
+        GitHub $github = null,
+        Split $split = null
+    ) {
+        parent::__construct($rootPath);
+
+        $this->rootPath = realpath($rootPath);
+        $this->components = $components;
+        $this->github = $github;
+        $this->split = $split;
+    }
+
+    protected function configure()
+    {
+        $this->setName('split')
+            ->setDescription('Split subtree and push to various remotes.')
+            ->addArgument(
+                'parent',
+                InputArgument::REQUIRED,
+                'The parent repository tag to which each component is linked.'
+            )
+            ->addOption(
+                'token',
+                't',
+                InputOption::VALUE_OPTIONAL,
+                sprintf(
+                    'A Github Auth Token. If not provided, uses value of environment variable `%s`.',
+                    self::TOKEN_ENV
+                )
+            )
+            ->addOption(
+                'splitsh',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Path to a pre-compiled splitsh-lite binary.'
+            )
+            ->addOption(
+                'component',
+                'c',
+                InputOption::VALUE_OPTIONAL,
+                'Split and release only a single component.'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (PHP_VERSION_ID < 50600) {
+            throw new \RuntimeException('This command is only available in PHP 5.6 and later.');
+        }
+
+        $execDir = $this->rootPath . '/' . self::EXEC_DIR;
+        $github = $this->github;
+        $split = $this->split;
+
+        $shell = new RunShell;
+
+        if (!$github) {
+            $output->writeln('<comment>[info]</comment> Instantiating GitHub API Wrapper.');
```

Comment:
This function is the crux of the command, so it's obviously got a lot going on. I think extracting some of these steps into other functions might help with organization. I would extract this bit that checks for a github instance, or instantiates the github into it's own function.

```
$shell = new RunShell;
$github = $this->githubClient();
```

Reply:
Good feedback! I can use that to stub the clients for testing as well in place of using constructor injection.